x-pack/plugins/fleet/public/applications/fleet/sections/epm/components/package_list_grid.tsx

```diff
@@ -129,17 +161,75 @@ function GridColumn({ list }: GridColumnProps) {
           </EuiFlexItem>
         ))
       ) : (
-        <EuiFlexItem>
+        <EuiFlexItem grow={3}>
           <EuiText>
             <p>
-              <FormattedMessage
-                id="xpack.fleet.epmList.noPackagesFoundPlaceholder"
-                defaultMessage="No packages found"
-              />
+              {showMissingIntegrationMessage ? (
+                <FormattedMessage
+                  id="xpack.fleet.epmList.missingIntegrationPlaceholder"
+                  defaultMessage="We didn't find any integrations matching your search term. Please try another keyword or browse using the categories on the left."
+                />
+              ) : (
+                <FormattedMessage
+                  id="xpack.fleet.epmList.noPackagesFoundPlaceholder"
+                  defaultMessage="No packages found"
+                />
+              )}
             </p>
           </EuiText>
         </EuiFlexItem>
       )}
     </EuiFlexGrid>
   );
 }
+
+interface MissingIntegrationContentProps {
+  resetQuery: () => void;
+  setSelectedCategory: (category: string) => void;
+}
+
+function MissingIntegrationContent({
+  resetQuery,
+  setSelectedCategory,
+}: MissingIntegrationContentProps) {
+  const history = useHistory();
+  const { getPath } = useLink();
+
+  const handleCustomInputsLinkClick = useCallback(() => {
+    resetQuery();
+    setSelectedCategory('custom');
+
+    history.push({
```

Comment:
Do we still need the `history.push()` if we only show this for all integrations?

Reply:
No, I don't think we do. We don't ever have to deal with the case of navigating to the "installed integrations" tab from this link now so this can be removed. Good catch.